sklearn/model_selection/_search.py

```diff
@@ -332,7 +332,7 @@ def _check_param_grid(param_grid):
             if isinstance(v, np.ndarray) and v.ndim > 1:
                 raise ValueError("Parameter array should be one-dimensional.")
 
-            check = [isinstance(v, k) for k in (list, tuple, np.ndarray)]
+            check = [isinstance(v, k) for k in (np.ndarray, Sequence)]
```

Comment:
I don't think we want to allow strings here. Strings are sequences.

Also, isinstance allows its second param to be a tuple of types, so `if isinstance(v, (np.ndarray, Sequence))` would suffice here... What we want is something like `if isinstance(v, (np.ndarray, Sequence)) and not isinstance(v, six.string_types)`. Please add a test